client/blocks/user-mentions/with-user-mentions.jsx

```diff
@@ -18,33 +22,250 @@ import React from 'react';
  * @returns {object} the enhanced component
  */
 export default EnhancedComponent =>
-	class withUserMentionSuggestions extends React.Component {
-		static displayName = `withUserMentionSuggestions( ${ EnhancedComponent.displayName ||
+	class withUserMentions extends React.Component {
+		matchingSuggestions = [];
+
+		static displayName = `withUserMentions( ${ EnhancedComponent.displayName ||
 			EnhancedComponent.name } )`;
 		static propTypes = {};
 
 		state = {
 			showPopover: false,
+			popoverContext: null,
+			popoverPosition: null,
+			query: '',
 		};
 
-		handleKeyPress = e => {
-			if ( e.target.value[ e.target.value.length - 1 ] === '@' ) {
-				console.log( 'found @something' ); // eslint-disable-line no-console
-				this.setState( { showPopover: true } );
+		constructor( props ) {
+			super( props );
+			// create a ref to store the textarea DOM element
+			this.textInput = React.createRef();
```

Comment:
Does `createRef` mark the textInput variable as a React reference?

Reply:
It does indeed.

https://reactjs.org/docs/refs-and-the-dom.html

Reply:
Oh! It's a pattern to replace the callback ref pattern. Got it :)